test/apps/iodemo/io_demo.cc

```diff
@@ -697,17 +740,28 @@ class DemoServer : public P2pDemoCommon {
         iov->init(msg->data_size, _data_chunks_pool, msg->sn, opts().validate);
         w->init(this, conn, msg->sn, iov, &_curr_state.write_count);
 
+        std::map<uint32_t, conn_state_t>::iterator it =
+                _conn_stat_map.find(conn->id());
+        assert(it != _conn_stat_map.end());
+        it->second.write_add(msg->data_size);
         recv_data(conn, *iov, msg->sn, w);
     }
 
     virtual void dispatch_connection_accepted(UcxConnection* conn) {
+        bool res = _conn_stat_map.insert(std::make_pair(conn->id(),
+                                                        conn_state_t())).second;
+        assert(res);
         ++_curr_state.active_conns;
     }
 
     virtual void dispatch_connection_error(UcxConnection *conn) {
         LOG << "deleting connection with status "
             << ucs_status_string(conn->ucx_status());
         assert(conn->is_established());
+        std::map<uint32_t, conn_state_t>::iterator it =
+                _conn_stat_map.find(conn->id());
+        assert(it != _conn_stat_map.end());
+        _conn_stat_map.erase(it);
```

Comment:
can we erase by key?

Reply:
we can but why? my point is similar to `insert` - `find` allows us to assert expected result and erase exact position